crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -555,7 +555,27 @@ impl<F: Clone + Send, Ctx: PaymentMethodRetrieve> Domain<F, api::PaymentsRequest
         requeue: bool,
         schedule_time: Option<time::PrimitiveDateTime>,
     ) -> CustomResult<(), errors::ApiErrorResponse> {
-        helpers::add_domain_task_to_pt(self, state, payment_attempt, requeue, schedule_time).await
+        // This spawns this futures in a background thread, the execption inside this future won't affect
+        // the current thread and the lifecycle of spawn thread is not handled by runtime.
+        // So when server shutdown won't wait for this thread's completion.
+        let m_payment_attempt = payment_attempt.clone();
+        let m_state = state.clone();
```

Comment:
This is an expensive call ideally we should have wrapper struct like
```rust
struct AppState {
	inner: Arc<ActualAppState>
}
```

Reply:
Yes, but touching `AppState` introduces lots of changes in the whole, and that changes can taking separately.

Reply:
Cool created Issue for the same https://github.com/juspay/hyperswitch/issues/2781